hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/opt/physical/index/IndexResolver.java

```diff
@@ -82,23 +89,18 @@ private IndexResolver() {
      * Analyzes the filter of the input scan operator, and produces zero, one or more {@link MapIndexScanPhysicalRel}
      * operators.
      *
-     * @param scan scan operator to be analyzed
+     * @param scan         scan operator to be analyzed
      * @param distribution distribution that will be passed to created index scan rels
-     * @param indexes indexes available on the map being scanned
+     * @param indexes      indexes available on the map being scanned
      * @return zero, one or more index scan rels
      */
     public static List<RelNode> createIndexScans(
```

Comment:
I would recommend adding isolated unit tests for `createIndexScans` to ensure that all the logic works as expected. It is difficult to assess the correctness because currently, we have only integration tests that check the final plan, but not the alternatives generated in MEMO. It should be relatively simple to create [scan, distribution, indexes] by hand and check which alternatives are returned.

Reply:
Addressed in https://github.com/hazelcast/hazelcast/pull/18013/commits/351d44b171ac38d0c560ab2fac985c3007a19787